MailManager/src/main/java/inc/softserve/resources/EnvelopeResource.java

```diff
@@ -21,25 +22,28 @@
 @Produces(MediaType.APPLICATION_JSON)
 public class EnvelopeResource {
 
-    private EnvelopeDao dao;
+    private EnvelopeDao envelopeDao;
+    private UserDao userDao;
     private EnvelopeTools envelopeTools;
 
     @Inject
-    public EnvelopeResource(EnvelopeDao dao, EnvelopeTools envelopeTools){
-        this.dao = dao;
+    public EnvelopeResource(EnvelopeDao envelopeDao, UserDao userDao, EnvelopeTools envelopeTools){
+        this.envelopeDao = envelopeDao;
+        this.userDao = userDao;
         this.envelopeTools = envelopeTools;
     }
 
     @POST
     @Timed
-    public long saveEnvelope(@QueryParam("Subject") String subject,
+    public String saveEnvelope(@QueryParam("Subject") String subject,
                                @QueryParam("To") String to,
                                @QueryParam("From") String username,
                                @QueryParam("template") String template,
                                @Context HttpServletRequest request){
-        Envelope envelope = new Envelope(EmailServiceApplication.users.get(username),
+        Envelope envelope = new Envelope(userDao.getUserByUsername(username),

```

Comment:
I'd assume that we would like to check whether user exists prior to creating and saving envelope